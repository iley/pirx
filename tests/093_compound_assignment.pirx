func main(): int {
  var x = 10;
  printf("x = %d\n", x);

  x += 5;
  printf("x += 5 -> %d\n", x);

  x -= 3;
  printf("x -= 3 -> %d\n", x);

  // Test with array indexing
  var arr = new([]int, 3);
  arr[0] = 100;
  arr[1] = 200;
  arr[2] = 300;

  printf("arr[1] = %d\n", arr[1]);
  arr[1] += 50;
  printf("arr[1] += 50 -> %d\n", arr[1]);
  arr[1] -= 25;
  printf("arr[1] -= 25 -> %d\n", arr[1]);

  dispose(arr);

  return 0;
}
