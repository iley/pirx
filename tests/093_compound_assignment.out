x = 10
x += 5 -> 15
x -= 3 -> 12
arr[1] = 200
arr[1] += 50 -> 250
arr[1] -= 25 -> 225
